pandas/core/indexes/multi.py

```diff
@@ -419,6 +419,11 @@ def _shallow_copy_with_infer(self, values=None, **kwargs):
     @Appender(_index_shared_docs['_shallow_copy'])
     def _shallow_copy(self, values=None, **kwargs):
         if values is not None:
+            # On equal MultiIndexes the difference is empty.
```

Comment:
this block can simply go in ``_shallow_copy_with_infer``, idea being we don't pass ``None`` to ``_shallow_copy`` unless we actually mean that.